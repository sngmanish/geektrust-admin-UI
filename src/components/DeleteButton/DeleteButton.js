import React from "react";

const DeleteButton = ({ anyBoxChecked, handleBatchDelete }) => {
  return (
    <button
      style={{
        width: "9rem",
        height: "2rem",
        color: "#7f8fa6",
        backgroundColor: "#dcdde1",
        border: "1px solid #7f8fa6",
        outline: "none",
        borderRadius: "1rem",
        ...(anyBoxChecked && {
          border: "none",
          backgroundColor: "#e74c3c",
          color: "#fff",
        }),
      }}
      disabled={!anyBoxChecked ? "disabled" : ""}
      onClick={handleBatchDelete}
    >
      Delete Selected
    </button>
  );
};

export default DeleteButton;
