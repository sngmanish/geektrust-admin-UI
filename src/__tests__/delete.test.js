import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import DeleteButton from '../components/DeleteButton/DeleteButton'

describe('DeleteButton', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<DeleteButton anyBoxChecked={false} handleBatchDelete={() => {}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('enables button when any box is checked', () => {
    const handleBatchDelete = jest.fn()
    render(<DeleteButton anyBoxChecked={true} handleBatchDelete={handleBatchDelete} />)
    const button = screen.getByText('Delete Selected')
    expect(button).not.toBeDisabled()
    fireEvent.click(button)
    expect(handleBatchDelete).toHaveBeenCalled()
  })

  it('changes style when any box is checked', () => {
    render(<DeleteButton anyBoxChecked={true} handleBatchDelete={() => {}} />)
    const button = screen.getByText('Delete Selected')
    expect(button).toHaveStyle(`
  border: 2px outset buttonface;
  background-color: #e74c3c;
  color: #fff;
`)
  })
})
