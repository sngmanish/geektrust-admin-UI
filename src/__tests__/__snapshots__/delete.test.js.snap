// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton renders correctly 1`] = `
<button
  disabled="disabled"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#dcdde1",
      "border": "1px solid #7f8fa6",
      "borderRadius": "1rem",
      "color": "#7f8fa6",
      "height": "2rem",
      "outline": "none",
      "width": "9rem",
    }
  }
>
  Delete Selected
</button>
`;
